### PR TITLE
Optimize the function of converting StoragePoolSpec to map

### PR DESCRIPTION
**Pull request by PengYiOfOpenSDS**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Solve Issue [#290](https://github.com/opensds/opensds/issues/290)
**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/controller/selector/filter_test.go [hunk 1]
```diff
@@ -379,3 +380,51 @@ func TestDiskTypeFilter(t *testing.T) {
...
+	Pool.Extras["mapA"] = mapA
+
+	result, err := GetPoolCapabilityMap(&Pool)
+	if nil != err {
+		t.Errorf("Expected nil, get %v", result)
+	}
+
+	CreatedAt, ok := result["createdAt"].(string)
+	if (!ok) || (Pool.CreatedAt != CreatedAt) {
+		t.Errorf("Expected true/%v, get %v/%v ", Pool.CreatedAt, ok, CreatedAt)
```
Comment (leonwanghui) [inline pkg/controller/selector/filter_test.go]: What do you mean in `/%v`?

File: pkg/controller/selector/filter_test.go [hunk 2]
```diff
@@ -379,3 +380,51 @@ func TestDiskTypeFilter(t *testing.T) {
...
+	}
+
+	thin, ok := result["extras.thin"].(bool)
+	if (!ok) || (Pool.Extras["thin"] != thin) {
+		t.Errorf("Expected true/%v, get %v/%v ", Pool.Extras["thin"], ok, thin)
+	}
+
+	value1, ok := result["extras.mapA.key1"].(string)
+	if (!ok) || ("value1" != value1) {
+		t.Errorf("Expected true/'value1', get %v/%v ", ok, value1)
```
Comment (leonwanghui) [inline pkg/controller/selector/filter_test.go]: Can you print the output of this command?

File: pkg/controller/selector/filter_test.go (hunk 1 above)
Comment (PengYiOfOpenSDS) [reply pkg/controller/selector/filter_test.go]: I compare 2 values, so separate them with /

File: pkg/controller/selector/filter_test.go (hunk 2 above)
Comment (PengYiOfOpenSDS) [reply pkg/controller/selector/filter_test.go]: This format is printable.

**Pull request merged**